api/src/pcapi/scripts/offer/copy_json_data_from_product.py

```diff
@@ -0,0 +1,155 @@
+    left_to_do = end_id - current
+    eta = left_to_do / batch_size * statistics.mean(elapsed_per_batch)
+    eta = datetime.datetime.utcnow() + datetime.timedelta(seconds=eta)
+    eta = eta.astimezone(pytz.timezone("Europe/Paris"))
+    eta = eta.strftime("%d/%m/%Y %H:%M:%S")
+    return eta
+
+
+def get_products_ids_from_file() -> list[str]:
+    products_ids = []
+    with open(FILE_CSV, newline="", encoding="utf-8") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=";")
+
+        for row in csv_reader:
+            products_ids.append(row[0])
+
+    return products_ids
+
+
+def divide_chunks(array: list, n: int) -> list:
+    for i in range(0, len(array), n):
+        yield array[i : i + n]
+
+
+def _fix_offers(batch_size: int, chunk_restart: int, max_chunk: int) -> None:
+    products_ids = get_products_ids_from_file()
+    logger.info("[copy_json_data_from_product] nb of products_ids : %d ", len(products_ids))
+    logger.info("[copy_json_data_from_product] batch_size : %d", batch_size)
+
+    chunks = divide_chunks(products_ids, batch_size)
+    nb_chunks = len(products_ids) / batch_size
+    logger.info("[copy_json_data_from_product] nb of chunks : %d ", nb_chunks)
+
+    elapsed_per_batch = []
+    # Nombre maximum de tentatives
+    max_attempts = 3
+    logger.info("[copy_json_data_from_product] max_attempts : %d", max_attempts)
+    i = 0
+    for ids in chunks:
+        if max_chunk and i == max_chunk:
+            break
+        if i < chunk_restart:
+            i = i + 1
+            continue
+        attempts = 0
+        start_time = time.perf_counter()
+        while attempts < max_attempts:
+            if (i % NB_LOGS) == 0:
+                logger.info("[copy_json_data_from_product] Tentative %d pour l'itération %d", (attempts + 1), i)
+            try:
+                ids_in = ",".join([str(id) for id in ids])
+                keys = [
+                    "author",
+                    "ean",
+                    "gtl_id",
+                    "rayon",
+                    "code_clil",
+                    "dewey",
+                    "titelive_regroup",
+                    "prix_livre",
+                    "schoolbook",
+                    "top",
+                    "collection",
+                    "num_in_collection",
+                    "comment",
+                    "editeur",
+                    "date_parution",
+                    "distributeur",
+                ]
+                for key in keys:
+                    key_replace = "'{" + key + "}'"
+                    sql = f"""
+                    UPDATE offer 
+                      SET "name" = product."name", 
+                          "description" = product."description", 
+                          "jsonData" = jsonb_set(
+                              COALESCE(offer."jsonData", product."jsonData"),
+                              {key_replace},
```

Comment:
```suggestion
                               {{ {key_replace} }},
```
Dans un f-string pour échaper les accolades il faut leur mettres des accolades.